Data Analysis Tools for Substrate-based Blockchains. (#1467)
w3f/Grants-Program: docs/RFPs/Open/data_analysis_tools.md

For example, Accounts has some provenance information that is pretty difficult or currently impossible to extract in block explorers. The account reference counter, account balance reserved provenance (see: https://docs.substrate.io/reference/account-data-structures/), and OpenGov delegations are examples of it. Some questions raised that use this data:

- Which transactions/accounts were responsible for the reserved balance in an account?
- What are the modules that are consumers, providers, and sufficient for a certain account and which transactions introduced/removed those references?
- Which accounts have delegated OpenGov votes to an account or to which accounts the account in question has delegated their votes to for each track, taking into account indirect delegations too (e.g. Account A delegates to Account B which delegates to Account C)?

This information is useful and requested for actual heavy users of the Polkadot/Kusama ecosystem.

> This RFP is not limited to the example above and intends to support other analyses. This RFP is not also limited to adding new features to the existent block explorer, the applicants can propose new analysis tools. Please notice that the intention here is not to create new block explorers that would have the same information, presented in the same way, as the current ones.

Reviewer: ```suggestion
This RFP is not limited to the example above and intends to support other analyses. This RFP is also not limited to adding new features to the existent block explorer, as applicants can propose new analysis tools as well. Please notice that the intention here is not to create new block explorers that would have the same information, presented in the same way, as the current ones.
```

## Deliverables :nut_and_bolt:

The expected deliverables are the tool features that provide specific data analysis. The data analysis provided by the tool should be detailed in the deliverables. Each analysis should be dynamic, reflecting the current state of the blockchain, and be presented in a web user interface, in a way that advanced non-technical users can consume, i.e., the user does not need to have programming skills. Please list each data analysis that will be supported by the tool in the deliverables including:

- The data analysis question. (ex: Which transactions were responsible to reserve the balance amount in an account?)
- The expected input for the data analysis (ex: an account)
- The expected output for the data analysis (ex: a set of transactions that made/removed a balance reserve in the input account)

> The proposed analysis should not overlap with the current ones existent if the information is easy to extract in block explores of Polkadot/Kusama ecosystem. They could overlap if the information is not simple or intuitively found by non-technical users in the current block explorers (ex. based on multiple steps in the block explorer or based on events data).

Reviewer: ```suggestion
The proposed analysis should not overlap with existing ones if the information is easy to extract in block explorers of the Polkadot/Kusama ecosystem. They can, however, overlap if the information is not simple or can't intuitively be found by non-technical users in the current block explorers (ex. based on multiple steps in the block explorer or based on events data).
```
